overlord/devicestate/devicemgr.go

```diff
@@ -1401,16 +1401,28 @@ func (m *DeviceManager) Unregister(opts *UnregisterOptions) error {
 			return err
 		}
 	}
+	oldKeyID := device.KeyID
 	device.Serial = ""
 	device.KeyID = ""
 	device.SessionMacaroon = ""
 	if err := m.setDevice(device); err != nil {
 		return err
 	}
-	// TODO: delete device keypair
+	// commit forgetting serial and key
+	m.state.Unlock()
+	m.state.Lock()
+	// delete the device key
+	err = m.withKeypairMgr(func(keypairMgr asserts.KeypairManager) error {
+		err := keypairMgr.Delete(oldKeyID)
```

Comment:
(nitpick^2) this could be a single line `if err := keypairMgr.Delete(); err != nil {...}`

Reply:
it's a small closure though so not sure it matters much either way